### PR TITLE
Add operator configuration

### LINKED ISSUE #113: Make the operator configurable
Right now we have a lot of constants the control how long the operator waits ofr various operations before deciding it is failed. This should be configurable because the latency can vary a lot between configurations.

### PR DESCRIPTION
**Pull request by voltbit**
- Refactored the constant wait times to be part of a config struct
- Replaced all generic timers with specific ones for each operation
- Added support to the ConfigMap controller for an operator config file
- Added support for auto loading configuration from a file on the disk

closes #113

### CODE REVIEW

**Comment by voltbit** [comment]
| Test | Time | Data/node |
| --- | --- | --- |
| TestRollingUpdate | 356.358s | 1.9GB |

tested on:
K8s `v1.19.11`
Linux `5.10.52`
`cgroupsV2`

File: controllers/config.go [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+				RedisManualFailoverCheckInterval:     2000 * time.Millisecond,
+				RedisManualFailoverCheckTimeout:      30000 * time.Millisecond,
+				RedisAutoFailoverCheckInterval:       2000 * time.Millisecond,
+				RedisAutoFailoverCheckTimeout:        30000 * time.Millisecond,
+			},
+		},
+	}
+}
+
+func (r *RedisOperatorConfig) LoadConfig() error {
```
Comment (shyimo) [inline controllers/config.go:184]: it's a bit confusing that this function is starting with Capital letter. this function is not being use outside of the package or even outside of the file. change it to lower case

File: controllers/config.go [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+	}
+	err = watcher.Watch(r.Path)
+	if err != nil {
+		r.Log.Error(err, fmt.Sprintf("Failed to watch config file at %s", r.Path))
+		return nil, err
+	}
+	return watcher, nil
+}
+
+func (r *RedisOperatorConfig) MonitorConfigFile() error {
```
Comment (shyimo) [inline controllers/config.go:211]: same as above

**Review by shyimo — COMMENTED** [review]
great job. really good feature !

**Review by shyimo — APPROVED** [review]

**Pull request merged**